libraries/AP_HAL_SITL/SITL_State_common.cpp

```diff
@@ -514,5 +514,47 @@ void SITL_State_Common::update_voltage_current(struct sitl_input &input, float t
     current2_pin_voltage = current_pin_voltage * .25f;
 }
 
+/*
+  set height above the ground in meters
+ */
+void SITL_State_Common::set_height_agl(void)
+{
+    static float home_alt = -1;
+
+    if (!_sitl) {
+        // in example program
+        return;
+    }
+
+    if (is_equal(home_alt, -1.0f) && _sitl->state.altitude > 0) {
+        // remember home altitude as first non-zero altitude
+        home_alt = _sitl->state.altitude;
+    }
+
+#if AP_TERRAIN_AVAILABLE
+    if (_sitl != nullptr &&
+        _sitl->terrain_enable) {
+        // get height above terrain from AP_Terrain. This assumes
+        // AP_Terrain is working
+        float terrain_height_amsl;
+        Location location;
+        location.lat = _sitl->state.latitude*1.0e7;
+        location.lng = _sitl->state.longitude*1.0e7;
+
+        AP_Terrain *_terrain = AP_Terrain::get_singleton();
+        if (_terrain != nullptr &&
+            _terrain->height_amsl(location, terrain_height_amsl, false)) {
```

Comment:
it would be better to move the height_agl variable to the sitl_fdm structure, as terrain DB is not available on periph

Reply:
Done